src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```
@@ -1054,6 +1057,96 @@ public void onNewRMarkdownDoc()
+                  }
+                  
+                  // Open and focus files that we created
+                  for (String filePath : JsUtil.asIterable(createdFiles.keys()))
+                  {
+                     FileSystemItem path = FileSystemItem.createFile(filePath);
+                     openFile(path, FileTypeRegistry.R, new CommandWithArg<EditingTarget>()
+                     {
+                        @Override
+                        public void execute(EditingTarget target)
+                        {
+                           target.setCursorPosition(Position.create(1, 0));
+                        }
+                     });
+                  }
+               }
+               
+               @Override
+               public void onError(ServerError error)
+               {
+                  Debug.logError(error);
+               }
+            });
+   }
+   
+   @Handler
+   public void onNewRShinyApp()
+   {
+      dependencyManager_.withShiny("Creating Shiny applications", new Command()
+      {
+         @Override
+         public void execute()
+         {
+            server_.getWorkingDirectory(new ServerRequestCallback<String>()
+            {
+               @Override
+               public void onResponseReceived(String workingDir)
+               {
+                  NewShinyWebApplication widget = new NewShinyWebApplication(
+                        "New Shiny Web Application",
+                        workingDir,
+                        new OperationWithInput<NewShinyWebApplication.Result>()
+                        {
+                           @Override
+                           public void execute(Result input)
+                           {
+                              doNewRShinyApp(input);
+                           }
+                        });
+
+                  widget.showModal();
+               }
+
+               @Override
+               public void onError(ServerError error)
```

I think you want to actually show there error to the user here (otherwise there will be a silent failure with no feedback). You can use the SimpleRequestCallback<T> class to automatically report the error via a standard onError handler.